plugins/linuxplugin/linuxplugin_init.go

```diff
@@ -47,10 +47,16 @@ type Plugin struct {
 	wg     sync.WaitGroup     // wait group that allows to wait until all goroutines of the plugin have finished
 }
 
-// GetIfIndexes gives access to mapping of logical names (used in ETCD configuration) to corresponding Linux interface indexes.
-// This mapping is especially helpful for plugins that need to watch for newly added or deleted Linux interfaces.
-func (plugin *Plugin) GetIfIndexes() idxvpp.NameToIdx {
-	return plugin.ifIndexes
+var (
+	// gPlugin holds the global instance of the Linux Plugin
+	gPlugin *Plugin
+)
+
+// GetLinuxIfIndexes gives access to mapping of logical names (used in ETCD configuration) to corresponding Linux
+// interface indexes. This mapping is especially helpful for plugins that need to watch for newly added or deleted
+// Linux interfaces.
+func GetLinuxIfIndexes() idxvpp.NameToIdxRW {
```

Comment:
I think it would be better, if we could avoid having global functions and use an instance method instead. (In a plugin where you want to use the indexes you have to just define the dependency on linux plugin and you're able to access the indexes.)